state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1748,61 +1908,19 @@ pub mod pallet {
 	}
 }
 
-impl<T: Config> SwapQueueApi for Pallet<T> {
-	type BlockNumber = BlockNumberFor<T>;
-
-	fn schedule_swap(
-		from: Asset,
-		to: Asset,
-		amount: AssetAmount,
-		refund_params: Option<SwapRefundParameters>,
-		swap_type: SwapType,
-	) -> (u64, Self::BlockNumber) {
-		let swap_id = SwapIdCounter::<T>::mutate(|id| {
-			id.saturating_accrue(1);
-			*id
-		});
-
-		// Do not limit the maximum swap amount for network fee swaps.
-		let swap_amount = if swap_type == SwapType::NetworkFee {
-			amount
-		} else {
-			let (swap_amount, confiscated_amount) = match MaximumSwapAmount::<T>::get(from) {
-				Some(max) => (sp_std::cmp::min(amount, max), amount.saturating_sub(max)),
-				None => (amount, Zero::zero()),
-			};
-			if !confiscated_amount.is_zero() {
-				CollectedRejectedFunds::<T>::mutate(from, |fund| {
-					*fund = fund.saturating_add(confiscated_amount)
-				});
-				Self::deposit_event(Event::<T>::SwapAmountConfiscated {
-					swap_id,
-					source_asset: from,
-					destination_asset: to,
-					total_amount: amount,
-					confiscated_amount,
-				});
-			}
-			swap_amount
-		};
-
-		let execute_at = frame_system::Pallet::<T>::block_number() + SWAP_DELAY_BLOCKS.into();
-
-		SwapQueue::<T>::append(
-			execute_at,
-			Swap::new(swap_id, from, to, swap_amount, refund_params, swap_type),
-		);
-
-		(swap_id, execute_at)
-	}
-}
-
 impl<T: Config> cf_traits::FlipBurnInfo for Pallet<T> {
 	fn take_flip_to_burn() -> AssetAmount {
 		FlipToBurn::<T>::take()
 	}
 }
 
+pub struct NoPendingSwaps<T: Config>(PhantomData<T>);
```

Comment:
If we introduce DCA, and a swap is spread over 12 hours or more, into its subswaps which are in the queue, then we can't submit the upgrade until it's done.  We'll likely want to migrate the swaps instead... but if we include this condition, even if we do choose to migrate them, we won't be able to push the upgrade to do so... and if we still allow contract swaps to come through someone could halt our ability to upgrade entirely... which would be near irrecoverable, since we would need to do an upgrade to reverse it.

Note also that this condition will *not* be applied to this current runtime upgrade, since we submit the upgrade using the extrinsic of the *old*/ current runtime.